backend/routes/items.js

```diff
@@ -101,26 +99,28 @@ router.post("/addItem", token_required, async (req, res, next) => {
     }).save();
   }
 
-  // Get the batch corresponding to this item and out date,
-  // creating a new batch if necessary.
-  let batch = await Batch.findOne({ itemName, outDate, batchId }).exec();
-  if (batch === null) {
-    batch = await new Batch({
-      itemName,
-      inDate,
-      outDate,
-      batchId,
-      poundsTotal: 0,
-      poundsRemaining: 0,
-    }).save();
+  // Ensure Batch is unique
+  let batchIdExist = await Batch.findOne({ batchId }).exec();
+  if (batchIdExist) {
+    return res.status(400).json({
```

Comment:
It looks like we're returning 400 in two separate cases: when `outDate` is in the past, and when the `batchId` already exists. I suppose the frontend could tell these cases apart by looking at the error message string, but that seems somewhat fragile (it'd be unintuitive if changing an error message string caused the frontend to stop working).

Could we add a `fieldName` attribute to the JSON object that will indicate which field is problematic? In this case, we could have `"fieldName": "outDate"` when the `outDate` is in the past, and `"fieldName": "batchId"` if the `batchId` is a duplicate. Since the field names are less likely to change, I feel like this would be more robust.

Reply:
Ooh, I see what you mean. I was thinking the frontend could just display the error message in a `<p>` tag or something to notify the user, kind of like how our authentication does right now. Let's bring this up with the team and see what they think